src/app/clusters/basic/basic.cpp

```diff
@@ -62,41 +62,251 @@ CHIP_ERROR BasicAttrAccess::Read(const ConcreteReadAttributePath & aPath, Attrib
         return CHIP_ERROR_INVALID_ARGUMENT;
     }
 
+    CHIP_ERROR status = CHIP_ERROR_INTERNAL;
+
     switch (aPath.mAttributeId)
     {
     case Location::Id:
-        return ReadLocation(aEncoder);
-    default:
+        status = ReadLocation(aEncoder);
+        break;
+
+    case VendorName::Id: {
+        constexpr size_t kMaxLen     = DeviceLayer::ConfigurationManager::kMaxVendorNameLength;
+        char vendorName[kMaxLen + 1] = { 0 };
+        status                       = ConfigurationMgr().GetVendorName(vendorName, sizeof(vendorName));
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(chip::CharSpan(vendorName, strnlen(vendorName, kMaxLen)));
+        }
```

Comment:
It seems to be the preference of the project to use the goto-wrapper and early return macros to reduce the number of dedicated error-checking if statements.

I don't know if any of those can be leveraged here.  Or perhaps a new macro could be introduced.  But there is a lot of this `if (status == CHIP_NO_ERROR) { // do next thing }` construct in this file.